public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  Display that array at the bottom of the component template with this `*ngFor` binding:
++makeExample('reactive-forms/ts/app/hero-detail.component.html', 'name-change-log','app/hero-detail.component.html (Name change log)')(format=".")
+
+:marked
+  Return to the browser, select a hero (e.g, "Magneta"), and start typing in the _name_ input box.
+  You should see a new name in the log after each keystroke.
+
+  ### When to use it
+  
+  An interpolation binding is the easier way to _display_ a name change.
+  Subscribing to an observable form control property is handy for triggering 
+  application logic _within_ the component class.
+
+.l-main-section
+a#save
+:marked
+  ## Save form data
+
+  The `HeroDetailComponent` captures user input but it doesn't do anything with it.
+  In a real app, you'd probably save those hero changes.
+  In a real app, you'd also be able to revert unsaved changes and resume editing.
+  After you implement both features in this section, the form will look like this:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/save-revert-buttons.png" width="389px" alt="Form with save & revert buttons")
+
+:marked
+  ### Save
+  In this sample application, the `HeroDetailComponent` cannot perform the save operation. 
+  The component delegates that chore to its parent `HeroListComponent`.
+  Althought not a rule, this is a common architectural design choice for reactive forms.
+  
+  When the user clicks the _Save_ button, the `HeroDetailComponent` raises a save event on its `save` output property.
+  The parent `HeroListComponent`, which binds to that property, responds to the event and processes the event payload.
++makeExample('reactive-forms/ts/app/hero-list.component.html', 'hero-binding','app/hero-list.component.html (save event binding)')(format=".")
+:marked
+  The save event payload _could_ have any shape but often it's in the shape of the data model and that is the case in this example.
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'save-emitter')(format=".")
+
+:marked
+  The `HeroDetailComponent` save method (`onSubmit`) constructs a `saveHero`
+  and emits a save message, with `saveHero` as the payload, for the `HeroListComponent` to process.
+
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'on-submit','app/hero-detail.component.ts (onSubmit)')(format=".")
+:marked
+  In the reactive paradigm, the _data model_ is immutable so the save method can't update the component's `hero`
+  and send that as the save event payload.
+
+  Instead, it must prepare `saveHero` whose values derive from a combination of original hero values (the `hero.id`)
+  and deep copies of the potentially-changed form model values.
+
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'prepare-save-hero','app/hero-detail.component.ts (prepareSaveHero)')(format=".")
+
+:marked
+  **Important**: do _not_ simply assign form model values to properties of `saveHero`.
```

Comment:
AFAIK ... there is no reason why a dev couldn't do this. If they don't care about immutability (or the "undo"), there is no reason they could not just assign form model properties to the 'saveHero'. It may be very misleading to have this here, especially as an important note.